tests/coords/transform_coords_test.py

```diff
@@ -175,11 +207,149 @@ def abcd(ac, bd):
     da = original.transform_coords(['abcd'], graph=graph)
     assert da.dims == ['ac', 'bd']
 
+    # c is also a dim coord
     original = sc.DataArray(data=a + b + c, coords={'a': a, 'b': b, 'c': c, 'd': b})
     da = original.transform_coords(['abcd'], graph=graph)
     assert da.dims == ['a', 'abcd', 'c']
 
 
+def test_dim_rename_multi_level_multi_merge_long():
+    # *x
+    #  |
+    # *a    *c
+    #  \   /
+    #    ac      *b
+    #     \     /
+    #      *abc
+    def abc(ac, b):
+        return ac + b
+
+    graph = {'a': 'x', 'ac': ac, 'abc': abc}
+    x = a.rename_dims({'a': 'x'})
+    original = sc.DataArray(data=x + b + c, coords={'x': x, 'b': b, 'c': c})
+    da = original.transform_coords(['abc'], graph=graph)
+    assert da.dims == ['a', 'abc', 'c']
+
+
+def test_rename_dims_cycle():
+    #   a
+    #   |
+    #   b   c
+    #  / \ /
+    # c   d  e
+    #  \ /  /
+    #   f __
+    def f_d(b, c):
+        return b + c
+
+    def f_f(c, d, e):
+        return c + d + e
+
+    graph = {'b': 'a', 'c': 'b', 'd': f_d, 'f': f_f}
+    original = sc.DataArray(data=a, coords={'a': a, 'c': a, 'e': a})
+    da = original.transform_coords(['f'], graph=graph)
+    assert da.dims == ['f']
+
+    # c is a dimension coord
+    original = sc.DataArray(data=a + c, coords={'a': a, 'c': c, 'e': a})
+    da = original.transform_coords(['f'], graph=graph)
+    assert da.dims == ['b', 'c']
+
+    # e is a dimension coord
+    e = a.rename_dims({'a': 'e'})
+    original = sc.DataArray(data=a + e, coords={'a': a, 'c': a, 'e': e})
+    da = original.transform_coords(['f'], graph=graph)
+    assert da.dims == ['b', 'e']
+
+
+def test_dim_rename_produced_dim_coord():
+    # c is a dimension coordinate even though it is computed from a and b.
```

Comment:
This is because it is introducing a new dimension and depends neither on the dims of `a` nor of `b`?

Reply:
I suppose this will fail if the data does not depend on that dim?